app/bundles/MessengerBundle/Middleware/SynchronousExtrasMiddleware.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+
+namespace Mautic\MessengerBundle\Middleware;
+
+use Mautic\MessengerBundle\MauticMessengerRoutes;
+use Mautic\MessengerBundle\Message\Interfaces\RequestStatusInterface;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Event\SendMessageToTransportsEvent;
+use Symfony\Component\Messenger\Middleware\MiddlewareInterface;
+use Symfony\Component\Messenger\Middleware\StackInterface;
+use Symfony\Component\Messenger\Stamp\ReceivedStamp;
+use Symfony\Component\Messenger\Stamp\SentStamp;
+use Symfony\Component\Messenger\Transport\Sender\SendersLocatorInterface;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * @author Jan Kozak <galvani78@gmail.com>
+ *
+ * Adds a stamp for request handling
+ */
+class SynchronousExtrasMiddleware implements MiddlewareInterface
```

Comment:
i have described the reason for this here: https://github.com/mautic/mautic/pull/12314#discussion_r1233723934

Reply:
@galvani based on the rebase and the latest changes I do not think this `Middleware` is needed, since messenger is processing based on the config, or is there something I am not getting?

Reply:
it is still needed. i allows setting a tracking cookie on synchronous processing.